### PR TITLE
Add spans for slow and frozen frames

### LINKED ISSUE #1912: Know when slow and frozen frames occurred during a transaction
### Description

Currently, the SDK attaches slow and frozen frames to transactions. The problem is that the users don't know when these slow and frozen occurred during the transaction. We could attach the slow and frozen frames to individual spans or we could add timing information as proposed in https://github.com/getsentry/sentry-cocoa/pull/1910, but then Sentry would need to do some work to add this somehow to the transactions. We could even add spans for slow and frozen frames.

Maybe only add to UI spans, because there could be two spans running at the same time. Furthermore, slow and frozen frames might not really make sense for file IO, DB, or network spans.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

When a slow or frozen frame occurs, all you see in the Sentry Profiling UI is a count of them, you don't know when they happened. With this change, you will be able to pinpoint when.

Screen Shot 2022-08-01 at 14 31 29

## :bulb: Motivation and Context

Closes #1912 

## :green_heart: How did you test it?

WIP

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [x] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryFramesTracker.m [hunk 1]
```diff
@@ -139,15 +144,20 @@ - (void)displayLinkCallback
     CFTimeInterval slowFrameThreshold = 1 / (actualFramesPerSecond - 1);
 
     CFTimeInterval frameDuration = lastFrameTimestamp - self.previousFrameTimestamp;
-    self.previousFrameTimestamp = lastFrameTimestamp;
```
Comment (kevinrenskers) [inline Sources/Sentry/SentryFramesTracker.m:142]: This was actually a bug in the existing code: by overwriting `self.previousFrameTimestamp` at this point, that meant that when we call the `recordTimestampStart` method below, the previous timestamp would always be equal to the `lastFrameTimestamp`. Both have the same value.

At the end of this method we're doing `self.previousFrameTimestamp = lastFrameTimestamp;` a second time so this first time is definitely not needed and seems to have been an oversight.

File: Sources/Sentry/SentryFramesTracker.m [hunk 2]
```diff
@@ -158,11 +168,32 @@ - (void)displayLinkCallback
     self.previousFrameTimestamp = lastFrameTimestamp;
 }
 
+- (void)sendTimestampSpanWithStart:(CFTimeInterval)start
+                               end:(CFTimeInterval)end
+                          isFrozen:(BOOL)isFrozen
+{
+    __block id<SentrySpan> frameSpan;
+    [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+        frameSpan = [span startChildWithOperation:@"frame.info"
```
Comment (kevinrenskers) [inline Sources/Sentry/SentryFramesTracker.m:177]: Not sure what a good operation name is here.

**Comment by kevinrenskers** [comment]
@brustolin when you look at the screenshot I added to the description, I would kinda expect that the frozen frame would appear inside of `viewAppearing`?

File: Sources/Sentry/SentryFramesTracker.m (hunk 2 above)
Comment (brustolin) [reply Sources/Sentry/SentryFramesTracker.m:177]: Im checking this with team.

File: Sources/Sentry/SentryFramesTracker.m (hunk 1 above)
Comment (brustolin) [reply Sources/Sentry/SentryFramesTracker.m:142]: Good catch

**Comment by brustolin** [comment]
> @brustolin when you look at the screenshot I added to the description, I would kinda expect that the frozen frame would appear inside of `viewDidLoad`?

Well, that depends on how you trigger the slow frame. If it was some code inside the viewDidLoad, so yes, it should be a child of 'viewDidLoad' span

**Comment by kevinrenskers** [comment]
I simply put a breakpoint inside viewDidLoad. And when you look at the screenshot, that is also the one that takes the longest.

**Comment by brustolin** [comment]
> I simply put a breakpoint inside viewDidLoad. And when you look at the screenshot, that is also the one that takes the longest.

I'll check. But dont worry, this is not an issue for this PR.

**Comment by brustolin** [comment]
After some investigation into this, it turns out that this is not adding any value and may cause some confusion. We are closing this for now. We can always revisit the idea later.

**Comment by philipphofmann** [comment]
> After some investigation into this, it turns out that this is not adding any value and may cause some confusion. We are closing this for now. We can always revisit the idea later.

Really, I'm a bit surprised. Why doesn't it add value to see where my slow and frozen frames occurred? Can you please elaborate, @brustolin?

**Comment by philipphofmann** [comment]
I just put some delays in an earlier lifecycle method of the UIViewController, and these spans make sense to me. I just pushed the changes. I pushed the changes, and I think we should revisit this PR.

Screen Shot 2022-08-30 at 17 50 27

**Comment by brustolin** [comment]
> Really, I'm a bit surprised. Why doesn't it add value to see where my slow and frozen frames occurred? Can you please.  elaborate, @brustolin?

The explanation is [here](https://github.com/getsentry/sentry-cocoa/pull/2028#issuecomment-1203716581), but the short version is because slow and frozen frames reports are a point in time and not a duration event, this alone make it no suitable to be a span. Besides that, during some calculations in the main thread, the report will be made at the end of the main loop, which does not help to identify the cause of the problem.

**Pull request closed without merging**